diff --git a/src/include/memory_chunk.h b/src/include/memory_chunk.h
index 3208242..e7df9d9 100644
--- a/src/include/memory_chunk.h
+++ b/src/include/memory_chunk.h
@@ -27,6 +27,7 @@
 #include <sys/stat.h>
 #include <unistd.h>
 #include <fcntl.h>
+#include <glib/gstdio.h>
 #include <stdlib.h>
 #ifdef HAVE_MMAP
 #include <sys/mman.h>
@@ -547,7 +548,7 @@ public:
             return false;
         }
 
-        fsync(fd);
+        g_fsync(fd);
         close(fd);
         return true;
     }
